src/app/pages/dashboard/types/widget-component.interface.ts

```diff
@@ -18,13 +18,8 @@ export type WidgetComponent<Settings extends SomeWidgetSettings = null> =
     ? WidgetComponentWithoutSettings
     : WidgetComponentWithSettings<Settings>;
 
-export interface WidgetSettingsComponent<Settings extends SomeWidgetSettings = null> {
-  // TODO: Figure out how to communicate with parent component to read and write settings and get validation state.
-  something: Settings;
-}
-
 type WidgetSettingsComponentType<Settings> = Settings extends SomeWidgetSettings
-  ? WidgetSettingsComponent<Settings>
+  ? Settings
```

Comment:
The original intent was to tie widget settings to widgets, so that developers don't make mistakes when they define their components.
This is now in broken state.

You need to:
1. Make `WidgetSettingsRef` accept a `Seettings` generic.
2. Recover back `WidgetSettingsComponent` and have it require `WidgetSettingsRef<Settings>` as one of the fields (instead of `something`).
3. Have `WidgetInterfaceIpSettingsComponent` implement `WidgetSettingsComponent` correctly.